src/recipes.jl

```diff
@@ -3,8 +3,24 @@
 # -------------------------------------------------
 # ## Dates & Times
 
-dateformatter(dt) = string(Date(Dates.UTD(dt)))
-datetimeformatter(dt) = string(DateTime(Dates.UTM(dt)))
+function epochdays2datetime(fractionaldays::Real)::DateTime
+    days = floor(fractionaldays)
+    dayfraction = fractionaldays - days
+    missing_ms = Millisecond(round(Millisecond(Day(1)).value * dayfraction))
+    DateTime(Dates.epochdays2date(days)) + missing_ms
+end
+
+epochdays2epochms(x) = Dates.datetime2epochms(epochdays2datetime(x))
+
+function dateformatter(dt::Integer)
+    string(Date(Dates.UTD(dt)))
+end
+
+function dateformatter(dt::Real)
+    string(DateTime(Dates.UTM(epochdays2epochms(dt))))
+end
```

Comment:
Why is one `Dates.UTM` and the other `Dates.UTD`?

Reply:
When using `Dates` in the data, values are converted to `epochdays` somewhere (I haven't found where...), which are integer number, so we can convert back to Dates via `UTD`.

However, when using e.g. `bar()` the bar-edges seem to be computed from the `epochdays`, causing some of them to fractional, which causes an error when trying to convert back to date. Thus, we convert `epochdays` -> `epochmilliseconds`, and use `UTM` to convert to `DateTimes`.
One could just `round()` the fractional `epochdays`, but then the barcenters and -edges would be the same `Date`, so the bar would become an infinitely thin vertical line.

Reply:
And that is not the case for `DateTimes` or should we handle it accordingly?

Reply:
For `DateTimes` we receive `epochmilliseconds` (as for any other format including `Time`), which are already handled correctly in the existing formatters.

We could just always handle `Dates` as `DateTime`, but existing functional might change, for which I am a bit hesitant (being new to this code base...)